### PR TITLE
Navigation setup

### LINKED ISSUE #2: Navbar component

### PR DESCRIPTION
**Pull request by andykwan1124**
Closes #2 

- Setup basic directory structure
- Setup 2 pages -> Home and Profile
- Setup routing btwn these 2 pages
- Created styled Navbar component

navbar

### CODE REVIEW

File: client/src/components/Navbar.tsx [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    display: flex;
+`
+
+const StyledLink = styled(Link)`
+    padding: 1em;
+    font-weight: bold;
+    color: black;
+    text-decoration: none;
+    font-size: 1.5em;
+    background-color: ${p => p.selected && '#A6B1A3'};
```
Comment (JohnTurkson) [inline client/src/components/Navbar.tsx:15]: I'm thinking we should use css variables for this (https://developer.mozilla.org/en-US/docs/Web/CSS/Using_CSS_custom_properties)

**Review by JohnTurkson — CHANGES_REQUESTED** [review]
Overall LGTM, apart from some small stylistic things that we should be consistent on:

- Can we use double quotes everywhere, including in imports?

File: client/tsconfig.json [hunk 2]
```diff
@@ -15,6 +15,7 @@
     "noFallthroughCasesInSwitch": true,
     "module": "esnext",
     "moduleResolution": "node",
+    "noImplicitAny": false,
```
Comment (JohnTurkson) [inline client/tsconfig.json:18]: I don't think this should be set to false. Setting this to true didn't yield any errors, and I don't think we should be reducing the capabilities of the ts compiler to check types in our project, which is likely going to lead to some avoidable errors.

File: client/package.json [hunk 3]
```diff
@@ -12,7 +12,9 @@
     "@types/react-dom": "^17.0.6",
```
Comment (JohnTurkson) [inline client/package.json:12]:
```suggestion
    "@types/react-dom": "^17.0.6",
    "@types/react-router-dom": "^5.1.7",
```

File: client/tsconfig.json (hunk 2 above)
Comment (JohnTurkson) [inline client/tsconfig.json:18]:
```suggestion
    "noImplicitAny": true,
```

File: client/src/components/Page.tsx [hunk 4]
```diff
@@ -0,0 +1,21 @@
+import Navbar from "./Navbar"
+import styled from 'styled-components'
+
+const MainContentContainer = styled.div`
+    margin-top: 5em;
+    padding: 1em;
+`
+
+const Page = ({ children }) => {
```
Comment (JohnTurkson) [inline client/src/components/Page.tsx:9]:
```suggestion
const Page = ({ children }: { children: ReactNode }) => {
```

File: client/src/components/Navbar.tsx [hunk 5]
```diff
@@ -0,0 +1,39 @@
+import { Link, useLocation } from 'react-router-dom'
+import styled from 'styled-components'
+
+const NavbarContainer = styled.div`
+    background-color: #B7C2B4;
+    display: flex;
+`
+
+const StyledLink = styled(Link)`
```
Comment (JohnTurkson) [inline client/src/components/Navbar.tsx:9]:
```suggestion
const StyledLink = styled(Link)<{ selected: boolean }>`
```

File: client/src/components/Page.tsx (hunk 4 above)
Comment (JohnTurkson) [reply client/src/components/Page.tsx:9]: https://stackoverflow.com/questions/53688899/typescript-and-react-children-type

File: client/src/components/Navbar.tsx (hunk 5 above)
Comment (JohnTurkson) [reply client/src/components/Navbar.tsx:9]: https://styled-components.com/docs/api#using-custom-props

**Comment by andykwan1124** [comment]
Personally I have no preference on single/double quotes, but I do see more ppl using single than double quotes in react apps. What do you think?

File: client/src/components/Navbar.tsx (hunk 1 above)
Comment (andykwan1124) [reply client/src/components/Navbar.tsx:15]: You mean to abstract out the hex color code for reusability? styled-components offers the ThemeProvider component, which we can use for storing common colors/styles. I can make that in a future PR

For ref: [https://styled-components.com/docs/advanced#theming](url)

**Comment by JohnTurkson** [comment]
> 
> 
> Personally I have no preference on single/double quotes, but I do see more ppl using single than double quotes in react apps. What do you think?

Let's go with double quotes.

**Review by JohnTurkson — APPROVED** [review]

**Pull request merged**